src/services/AuthService.js

```diff
@@ -14,34 +17,50 @@ class AuthService {
 
   init = () => {
     this.setAuthToken(this.getAccessToken());
-    this.httpService.addRequestInterceptor();
+    this.httpService.addRequestInterceptor(this.checkTokenExpiration);
     this.httpService.addResponseInterceptors();
   };
 
   getAccessToken = () => {
-    return JSON.parse(localStorage.getItem("token"));
+    return {
+      token: getItem("token"),
+      refresh: getItem("refresh"),
+    };
   };
 
-  setAuthToken = (token) => {
+  setAuthToken = ({ token, refresh }) => {
     if (token) {
-      localStorage.setItem("token", JSON.stringify(token));
+      setItem("token", token);
+      setItem("refresh", refresh);
 
       this.httpService.attachHeaders({
         Authorization: `Bearer ${token}`,
       });
     }
   };
 
-  login = async (data) => {
-    const { access: token } = await this.httpService.request({
+  refreshToken = async (refresh) => {
+    const { data } = await this.httpService.request({
+      url: ROUTES.REFRESH,
+      method: "POST",
```

Comment:
Neka ovo bude u constants fajlu kao sve metode na jednom mestu, pa to importuj i koristi ovde  `HTTP_METHODS.POSTS` npr
Ako ima jos negde da je ovako string korisceno, zameni svuda